main.go

```diff
@@ -297,6 +410,13 @@ func _main() error {
 				}
 				i += 1
 
+				// stdin has been exhausted after hashing
+				if cache {
+					b, err = io.ReadAll(projectTmp)
```

Comment:
ReadAll is a bad sign. Why are you doing this?

Reply:
I think it would be better to edit like 421 and and wherever `b` is first defined have it also do:

```
var r io.Reader = bytes.NewReader(b)
```

so that here you can do something like 

```
if cache {
  r, err = io.Open(projectTmp)
}
```

And not have to read the whole thing twice

Reply:
Because the existing code is also ReadAll from os.Stdin, the projectTmp file is just a copy of os.Stdin in this case. So should I remove both  readAlls?

Reply:
Take a look at my second comment and see if that makes sense